var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -318,6 +318,8 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+strumpack", when="@:13.0")
     # Can only use one type of SuperLU
     conflicts("+superlu-dist", when="+superlu")
+    conflicts("+amesos", when="+superlu")
+    conflicts("+ifpack", when="+superlu")
```

Comment:
```suggestion
    conflicts("+amesos", when="+superlu@:4.3")
    conflicts("+ifpack", when="+superlu@:4.3")
```
Should we consider embedding this knowledge by making the conflict conditional on the version rather than just turning it off? Would this be the right version?

Reply:
We would need `when="+superlu@5.2.1:"` to turn off the packages when using versions of superlu at 5.2.1 and above.

Reply:
```suggestion
    conflicts("+amesos", when="+superlu@5:")
    conflicts("+ifpack", when="+superlu@5:")
```

It should probably be something like this. If an older superlu is enabled then these old packages should still build with it. These are going to go away eventually when epetra is removed, right?